### ISSUE TITLE
V2: Verify postal webhook signature

### ISSUE THREAD

**Issue by Josh-G** [SynergiTech/laravel-postal#5]
we should add functionality to verify the `X-Postal-Signature` webhook header

**Comment by willpower232** [comment]
Code reference:
```php
$dkim_public_key = '...'; // Just the p=... part of the TXT record (without the semicolon at the end)
$rsa_key_pem = "-----BEGIN PUBLIC KEY-----\r\n" .
    chunk_split($dkim_public_key, 64) .
    "-----END PUBLIC KEY-----\r\n";
$rsa_key = openssl_pkey_get_public($rsa_key_pem);

$body = file_get_contents('php://input');

$signature = $_SERVER['HTTP_X_POSTAL_SIGNATURE'];
$signature = base64_decode($signature);

$result = openssl_verify($body, $signature, $rsa_key, OPENSSL_ALGO_SHA1);

if ($result === 1) {
    // Verified!
}
```

Based on this, we'll need to start storing the Postal public key of each domain we're sending from as config in the project (rather than querying for TXT records on the fly) but [apparently](https://stackoverflow.com/questions/53415485/laravel-cant-get-pem-public-key-data-from-env-file) the .env files won't store multi line values.

I guess we should just store the stated component of the DKIM record and do the `chunk_split` as above all the time?

The rest shouldn't be difficult to laravel'ify.

**Comment by willpower232** [comment]
Resolved in #4
